clis/teliod/src/main.rs

```diff
@@ -44,6 +46,9 @@ enum Cmd {
     Daemon { config_path: String },
     #[clap(flatten)]
     Client(ClientCmd),
+    #[cfg(feature = "qnap")]
+    #[clap(about = "Receive and parse http requests")]
+    Cgi,
```

Comment:
I don't get it tbh - why is it added here?  It seems it communicates with the rest of the teliod code through cli interface, it looks like completely separate tool, to me it seems that there is no reason to keep it together.

Reply:
you're right it could have been a separate tool. The team has decided the pros didn't outweight the cons and decided to have this design. What's your rationale in splitting it into a different binary?

Reply:
Well, it doesn't really share any code, I see no arguments for bundling it together. The cons are quite obvious, you create false impression that these pieces of code are somehow connected, you cannot compile them separately (when compiling for qnap), but most importantly, as function shouldn't do job of two function based on some boolean, a binary shouldn't be, in fact, two separate binaries, glued with some `cgi` flag.

Reply:
I think you can use the same reasoning for the teliod as a daemon or client. It was decided to keep them under the same binary, and the same decision was made for qnap's cgi.

> you cannot compile them separately

Why would you insist on having them separated? There's this CGI feature that is only used for qnap usecase and therefore excluded for other platform builds.

> as function shouldn't do job of two function based on some boolean, a binary shouldn't be, in fact, two separate binaries, glued with some cgi flag.

What is the job of two functions? We call qnap::parse_cgi() to handle the http request, that's one usecase only. And this isn't a function that's enabled by a boolean, it's included or excluded from the binary depending on the flag you pass to the compiler at compile-time so there's no code/symbols on binaries for other platforms. That's what cargo features are about, it's literally used by almost all crates. 🤔

Reply:
>I think you can use the same reasoning for the teliod as a daemon or client

TBH if I would start from scratch, I probably would make them separate, but they actually share some code and communicate in some reasonable manner, while this new part treats the rest of teliod like a blackbox

>What is the job of two functions? We call qnap::parse_cgi() to handle the http request, that's one usecase only. And this isn't a function that's enabled by a boolean, it's included or excluded from the binary depending on the flag you pass to the compiler at compile-time so there's no code/symbols on binaries for other platforms. That's what cargo features are about, it's literally used by almost all crates. 🤔

I'm not talking about compilation flag, but about `cgi` parameter - you pass it and run some program which is completely disjoint with the rest. I'm not insisting on separating them, but I think there are two reasonable options:
 1. Integrate it, so use the socket, use consistent checks of the stuff, errors, etc.
 2. Keep it as a separate script like now, but in such case bundling it together has no benefits imho

Reply:
Yeh this came down via vote, and all cases are equivalent.

But I would be for not calling `teliod client` from `teliod cgi` but actualy reasuing socket logic to interact with deamon directly, like teliod client.

Metal model being that there are two client impl's for teliod daemon:
- cli interface
- web-cgi interface

Reply:
Added socket logic instead of issuing the bash command.